### PR TITLE
Crypt Package: Add Exception to Docblock

### PR DESCRIPTION
**Pull request by AmyStephen**

### CODE REVIEW

File: libraries/joomla/crypt/key.php [hunk 1]
```diff
@@ -74,7 +75,7 @@ public function __get($name)
 		}
 		else
 		{
-			trigger_error('Cannot access property ' . __CLASS__ . '::' . $name, E_USER_WARNING);
+			throw new BadMethodCallException('Cannot access property ' . __CLASS__ . '::' . $name);
```
Comment (realityking) [inline libraries/joomla/crypt/key.php]: Not sure about this one. I suspect the idea was to mimic the behavior of PHP when the property doesn't exist. With your change I'd have to add a try catch (while catching BadMethodCallException) around simple property accesses like $key->something. That seems weird to me and would probably be more confusing than helpful.

Comment (AmyStephen) [inline libraries/joomla/crypt/key.php]: Well, the other thing is, I really don't think this logic can ever be reached given how the object class is identified as a property of the method call parameter.

I'll go ahead and remove it based on your point.

File: libraries/joomla/crypt/key.php [hunk 2]
```diff
@@ -65,6 +65,7 @@ public function __construct($type, $private = null, $public = null)
 	 * @return  mixed
 	 *
 	 * @since   12.1
+	 * @throws  BadMethodCallException
```
Comment (realityking) [inline libraries/joomla/crypt/key.php]: You forgot the docblock.

BTW the logic is reached when I try to access some property that just doesn't exist.

**Comment by realityking** [comment]
Thanks Amy!

**Pull request merged**